intermediate_source/_torch_export_nightly_tutorial.py

```diff
@@ -0,0 +1,567 @@
+# ``dynamic_shapes_example3``, the exported program is sound even though there is a
+# raw ``if`` statement.
+#
+# If you want to see why ``torch.export`` generated these constraints, you can
+# re-run the script with the environment variable ``TORCH_LOGS=dynamic,dynamo``,
+# or use ``torch._logging.set_logs``.
+
+import logging
+torch._logging.set_logs(dynamic=logging.INFO, dynamo=logging.INFO)
+exported_dynamic_shapes_example3 = export(dynamic_shapes_example3, (inp4, inp5), dynamic_shapes=dynamic_shapes3_fixed)
+
+# reset to previous values
+torch._logging.set_logs(dynamic=logging.WARNING, dynamo=logging.WARNING)
+
+######################################################################
+# We can view an ``ExportedProgram``'s constraints using the ``range_constraints`` and
+# ``equality_constraints`` attributes. The logging above reveals what the symbols ``s0, s1, ...``
+# represent.
+
+print(exported_dynamic_shapes_example3.range_constraints)
+print(exported_dynamic_shapes_example3.equality_constraints)
+
+######################################################################
+# We can also constrain on individual values in the source code itself using
+# ``constrain_as_value`` and ``constrain_as_size``. ``constrain_as_value`` specifies
+# that a given integer value is expected to fall within the provided minimum/maximum bounds (inclusive).
+# If a bound is not provided, then it is assumed to be unbounded.
+
+from torch.export import constrain_as_size, constrain_as_value
+
+def dynamic_shapes_example4(x, y):
+    b = y.item()
+    constrain_as_value(b, 3, 5)
+    if b >= 3:
+       return x.cos()
+    return x.sin()
+
+exported_dynamic_shapes_example4 = export(dynamic_shapes_example4, (torch.randn(3, 3), torch.tensor([4])))
+print(exported_dynamic_shapes_example4(torch.randn(3, 3), torch.tensor([5])))
+try:
+    exported_dynamic_shapes_example4(torch.randn(3, 3), torch.tensor([2]))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# ``constrain_as_size`` is similar to ``constrain_as_value``, except that it should be used on integer values that
+# will be used to specify tensor shapes -- in particular, the value must not be 0 or 1 because
+# many operations have special behavior for tensors with a shape value of 0 or 1.
+
+def dynamic_shapes_example5(x, y):
+    b = y.item()
+    constrain_as_size(b)
+    z = torch.ones(b, 4)
+    return x.sum() + z.sum()
+
+exported_dynamic_shapes_example5 = export(dynamic_shapes_example5, (torch.randn(2, 2), torch.tensor([4])))
+print(exported_dynamic_shapes_example5(torch.randn(2, 2), torch.tensor([5])))
+try:
+    exported_dynamic_shapes_example5(torch.randn(2, 2), torch.tensor([1]))
+except Exception:
+    tb.print_exc()
+
+######################################################################
+# Custom Ops
+# ----------
+#
+# ``torch.export`` can export PyTorch programs with custom operators.
```

Comment:
can you also link https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU/edit#heading=h.ahugy69p2jmz